dashboard/src/components/api/che-workspace.factory.js

```diff
@@ -334,26 +334,18 @@ export class CheWorkspace {
    */
   getWebsocketUrl(workspaceId) {
     let workspace = this.workspacesById.get(workspaceId);
-    if (!workspace || !workspace.runtime) {
+    if (!workspace || !workspace.runtime || !workspace.runtime.links) {
       return '';
     }
-    let runtimeData = workspace.runtime;
 
-    // extract the Websocket URL of the runtime
-    let servers = runtimeData.devMachine.runtime.servers;
-
-    var wsagentServerAddress;
-    for (var key in servers) {
-      let server = servers[key];
-      if ('wsagent' === server.ref) {
-        wsagentServerAddress = server.address;
+    for (let i = 0; i < workspace.runtime.links.length; i++) {
+      let link = workspace.runtime.links[i];
+      if (link.rel === 'wsagent.websocket') {
+        let href = link.href.endsWith('/') ? link.href : link.href + '/';
+        return href + 'ws/' + workspaceId;
```

Comment:
I think that client must not know the path to the websocket endpoint. Please move that into link on serverside